clean.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b     Delete bin directory"
+    echo "  -p     Delete packages directory and NuGet package caches"
+    echo "  -a     Remove all untracked files in the working tree and delete NuGet package caches"
+    exit 1
+}
+
+# Parse arguments
+
+if [ $# == 0 ]
+then
+    echo "Please specify at least one option."
+    echo "Use 'clean -h' for help."
+    exit 1
+fi
+
+while [[ $# > 0 ]]
+do
+    opt="$1"
+    case $opt in
+        -h|--help)
+        usage
+        ;;
+        -b)
+        CleanBinDir=true
+        ;;
+        -p)
+        CleanPackagesDir=true
+        CleanPackagesCache=true
+        ;;
+        -a)
+        CleanWorkingTree=true
+        CleanPackagesCache=true
+        ;;
+        *)
+        echo "Unrecognized argument '$opt'"
+        echo "Use 'clean -h' for help."
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+WorkingTreeRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ "$CleanBinDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing bin directory"
+    rm -rf $WorkingTreeRoot/bin
+fi
+
+if [ "$CleanPackagesDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing packages directory"
+    rm -rf $WorkingTreeRoot/packages
+fi
+
+if [ "$CleanWorkingTree" == true ]
+then
+    echo "Removing all untracked files in the working tree"
+    git clean -xdf $WorkingTreeRoot
+fi
+
+if [ "$CleanPackagesCache" == true ]
+then
+    echo "Removing ~/.local/share/NuGet/Cache/"
+    rm -rf ~/.local/share/NuGet/Cache
+    echo "Removing ~/.local/share/NuGet/v3-cache/"
+    rm -rf ~/.local/share/NuGet/v3-cache
+    echo "Removing ~/.local/share/dnu/cache/"
+    rm -rf ~/.local/share/dnu/cache
+fi
```

Comment:
I think it makes sense to have a separate option for  deleting the user-local package caches.  I'll make a "-c" option for that, so it will only be done when using "-c" or "-a".